lms/djangoapps/courseware/masquerade.py

```diff
@@ -154,7 +154,7 @@ def is_masquerading_as_specific_student(user, course_key):  # pylint: disable=in
     Returns whether the user is a staff member masquerading as a specific student.
     """
     course_masquerade = get_course_masquerade(user, course_key)
-    return bool(course_masquerade and course_masquerade.user_name)
+    return bool(course_masquerade and getattr(course_masquerade, 'user_name', None))
```

Comment:
@smarnach the rest of the code in this file is written under assumption that if masquerade record exists it is a `CourseMasquerade` instance, thus having all the attributes and methods of it. If I do understand right, the issue existed because there was already a record with old `CourseMasquerade` structure (without `user_name`), and somehow it was deserialized into an object that also did not have `user_name`.

It would be a common case if more fields are added to `CourseMasquerade`, so it might make sense to add `__getattr__` method to log warning/error and return `None` in case of missing attribute - that would protect future generations of having to always use `getattr` instead of normal attribute access. It also might not work at all, but I believe it is worth trying at least.